desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphLinkPainter.java

```diff
@@ -289,6 +295,36 @@ public void actionPerformed(ActionEvent e) {
         return false;
     }
 
+    /**
+     * This will check if components are in a different scope, and ask the user for
+     * permission to change the scope of the target component
+     *
+     * @return true if permitted or irrelevant, false if user refused a necessary scope change.
+     */
+    private boolean scopeUpdatePermitted(final AnalysisJobBuilder sourceAnalysisJobBuilder,
+            final ComponentBuilder componentBuilder) {
+        if (sourceAnalysisJobBuilder != componentBuilder.getAnalysisJobBuilder()) {
+            if (componentBuilder.getInput().length > 0 || componentBuilder.getComponentRequirement() != null) {
+                final String scopeText;
+                if(sourceAnalysisJobBuilder.isRootJobBuilder()){
+                    scopeText = " into the default scope";
+                } else {
+                    scopeText = " into the scope " + sourceAnalysisJobBuilder.getDatastore().getName();
+                }
+                final int response = JOptionPane.showConfirmDialog(_graphContext.getVisualizationViewer(),
+                        "This will move " + componentBuilder.getDescriptor().getDisplayName() + scopeText
```

Comment:
Instead of .getDisplayName() let's use LabelUtils.getLabel(componentBuilder). That will respect also if the user has done any renaming etc.